RedisEnterprise-DNS-Migration/migration_script.ps1

```diff
@@ -0,0 +1,325 @@
+            Write-Host "Confirming zone has the correct A-Records added..." -ForegroundColor Yellow
+            foreach ($privateLink in $privateLinks) {
+                $privateDnsZoneConfig = New-AzPrivateDnsZoneConfig -Name ($newZone.Name -Replace "\.", "-") -PrivateDnsZoneId $newZone.ResourceId
+                $shouldProceed = $false
+                $isDisconnected = $privateLink.PrivateLinkServiceConnections.PrivateLinkServiceConnectionState.Status.toLower() -eq 'disconnected'
+                if(!$isDisconnected) {
+                    $existingGroup = Get-AzPrivateDnsZoneGroup -ResourceGroupName $privateLink.ResourceGroupName -PrivateEndpointName $privateLink.Name
+                    if($null -ne $existingGroup -and $existingGroup.Count -gt 0) {
+                        
+                        if (!$shouldProceed -and $null -eq $existingGroup.PrivateDnsZoneConfigs) {
+                            
+                            Write-Host "[Step 4 under section 'How to migrate to the new private DNS zone' in migration document]" -ForegroundColor Yellow
+                            $shouldProceed = AskForConfirmation "Private endpoint '$($privateLink.Name)' is already tied to a DNS zone group (A-Record) which has no associated DNS zones. Do you want to remove the existing group and add a new one?"
+                        }
+                        if (!$shouldProceed -and !($existingGroup.PrivateDnsZoneConfigs.PrivateDnsZoneId.ToLower() -contains $newZone.ResourceId.toLower())) {
+                            
+                            Write-Host "[Step 4 under section 'How to migrate to the new private DNS zone' in migration document]" -ForegroundColor Yellow
+                            $shouldProceed = AskForConfirmation "Private endpoint '$($privateLink.Name)' is already tied to a DNS zone group (A-Record) that is not associated with this DNS Zone. Do you want to remove the existing group and add a new one?"
+                        }
+                        $peIpAddresses = ((Get-AzNetworkInterface -ResourceId $privateLink.NetworkInterfaces.Id).IpConfigurations.PrivateIPAddress)
+                        $zoneARecordIpAddresses = ($newZone | Get-AzPrivateDnsRecordSet | Where-Object RecordType -eq A).Records.Ipv4Address
+                        if (!$shouldProceed -and !($zoneARecordIpAddresses -contains $peIpAddresses)) {
+                            
+                            Write-Host "[Step 4 under section 'How to migrate to the new private DNS zone' in migration document]" -ForegroundColor Yellow
+                            $shouldProceed = AskForConfirmation "Private endpoint '$($privateLink.Name)' has an IP address of $($peIpAddresses) which is not contained in the private DNS zones list of A records which include [$($zoneARecordIpAddresses -join ", ")]? Do you want to and add a new one dns zone group to fix this?"
```

Comment:
Candidate little typo fixes for the second sentence, and just to brainstor, a possible wording idea for the first sentence:

```suggestion
                            $shouldProceed = AskForConfirmation "Private endpoint '$($privateLink.Name)' has an IP address of $($peIpAddresses), which is not in the private DNS zone's list of A records: [$($zoneARecordIpAddresses -join ", ")]? Do you want to add a new DNS zone group to the private endpoint to fix this?"
```